src/qutip_qip/operations/gates.py

```diff
@@ -1232,6 +1233,9 @@ def expand_operator(
         E.g. if ``N=3`` and `oper` is a 2-qubit operator,
         the result will be a list of three operators,
         each acting on qubits 0 and 1, 1 and 2, 2 and 0.
+    dtype : str, optional
+        Data type of the output `Qobj`. Only for qutip version larger than 5.
```

Comment:
In https://github.com/qutip/qutip/pull/2321 the QuTiP creation operators are currently being updated to ensure that if a `dtype` is specified, the returned Qobj has the requested type, even if the dispatch logic picks something else, because, e.g., a data layer is missing its own implementation of some operation.

It would be nice to be consistent here.

The mentioned PR starts such functions with `dtype = dtype or settings.core["default_dtype"] or _data.CSR` and maybe that is good for this case too? It would mean that the dtype is always CSR unless specified, but perhaps that is better than the current more complex behaviour? Thoughts?

Reply:
Oh thanks, this looks good, especially using the `settings.core["default_dtype"]`.

The point here is to use sparse as much as possible when the input is `Dense`. But it does make sense that if a default data type is specified, one should use that.